File: backend/TextileOasis.Domain/Entities/Order.cs
namespace TextileOasis.Domain.Entities;

public class Order
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public decimal Total { get; set; }
    public decimal ShippingCost { get; set; }
    public string Status { get; set; } = "Pending";
    public string CustomerFullName { get; set; } = string.Empty;
    public string CustomerEmail { get; set; } = string.Empty;
    public string CustomerPhoneNumber { get; set; } = string.Empty;
    public string ShippingAddress { get; set; } = string.Empty;
    public string ShippingDistrict { get; set; } = string.Empty;
    public string ShippingCity { get; set; } = string.Empty;
    public string DeliveryType { get; set; } = "delivery";
    public string PickupStoreAddress { get; set; } = string.Empty;
    public string PaymentMethod { get; set; } = "cash";
    public string PaymentStatus { get; set; } = "Pending";
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();
}

File: src/types/order.ts
export interface CreateOrderItemDto {
  productId: number;
  quantity: number;
}

export interface CreateOrderDto {
  items: CreateOrderItemDto[];
  customerFullName: string;
  customerEmail: string;
  customerPhoneNumber: string;
  shippingAddress: string;
  shippingDistrict: string;
  shippingCity: string;
  deliveryType: "pickup" | "delivery";
  paymentMethod: "card" | "transfer" | "cash";
  shippingCost: number;
}

export interface OrderItem {
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
  subTotal: number;
}

export interface Order {
  id: number;
  createdAt: string;
  total: number;
  shippingCost: number;
  status: string;
  customerFullName: string;
  customerEmail: string;
  customerPhoneNumber: string;
  shippingAddress: string;
  shippingDistrict: string;
  shippingCity: string;
  deliveryType: string;
  pickupStoreAddress: string;
  paymentMethod: string;
  paymentStatus: string;
  items: OrderItem[];
}
